### DISCUSSION TITLE
Vorschläge Konfigurationsdialog, Migrationshilfe, Anpassung Dokumenation

### DISCUSSION THREAD

**Discussion by hklages** [SukramJ/aiohomematic#441] [Ideas]
1. Der Konfigurationsdialog besteht aus 2 Seiten. Dies ist leider nicht erkenntlich. 
Vorschlag: Irgendwo Seite 1 von 2 angeben oder statt "submit" besser "next page" auf der ersten Seite

2. Gibt es irgendwo einen "Platz" für Migrationstipps? Im Wiki? Oder hier als Diskussionpunkt, der "pinned" ist
Beispiel: Umstellung von "Listen to Events" (param, address versus parameter and device_id)
Beispiel: Nutzung von set_device_value
Beispiel: Hinweis auf Blueprints für Buttons

**Comment by SukramJ** [comment]
zu 1. Der ConfigFlow wird nur mit Werten bestückt, das ganze Umfeld kommt von HA und ist relativ eingeschränkt, zumindest wenn man nicht auch im HA Frontend etwas ändert. So weit ich weiss kann man die Schaltflächen nicht austauschen

zu 2. Klar, das Wiki hat Daniel zum Bearbeiten für Jederman geöffnet, aber wie Du siehst ist die Beteiligung mau. 
Also besteht hier für dich die Möglichkeit den Grundstein zu legen.

Da du ja bestimmt die [Doku](https://github.com/danielperna84/custom_homematic/blob/devel/README.md) gelesen hast sind dir bestimmt auch die [Beispiele](https://github.com/danielperna84/custom_homematic/blob/devel/README.md#examples-in-yaml) aufgefallen. Weitere Beispiele kann die Community natürlich im Wiki beisteuern.

Die [Blueprints](https://github.com/danielperna84/custom_homematic/blob/devel/README.md#available-blueprints) sind natürlich auch in der Doku erwähnt.

**Comment by hklages** [comment]
Danke - ja, die Dokumentation ist bereits sehr gut (und die Integration exzellent). 
2 kleine Änderungen haben ich in eine pull request gepackt. Die device_id hat mich etwas Zeit gekostet. Erst hatte ich es übersehen und weiterhin address verwendet und dann wusste ich nicht, wie man sie findet.

**Reply by SukramJ** [reply]
Hier sind meine Gedanken zu device_id mit Bezug zu [PR](https://github.com/danielperna84/custom_homematic/pull/223):

Ich glaube, hier liegt ein konzeptionelles Missverständnis vor, was sich auch in den hier gezeigten YAML-Beispielen widerspiegelt.
Die device_id ist eine interne Kennung, die von HA für UI-erstellte Automatisierungen für gerätebasierte Trigger und Aktionen verwendet wird.
Aus diesem Grund ist diese Kennung nicht leicht zugänglich.

Wenn man Automatisierungen manuell erstellt, kann man entweder die Grundstruktur über die UI und die Logik manuell in YAML hinzufügen, oder man sollten ereignisbasierte Trigger verwenden.
Die Daten für ereignisbasierte Trigger können entweder aufgezeichnet oder von den Entitäten gelesen werden.

@danielperna84 Ich bin mir nicht sicher, wie man in der Dokumentation mit manuellen/UI-basierten Automatisierungen umgehen sollte. Ich denke hier wäre eine Klarstellung nötig. Teile der Automatisierungsdokumentation sollten auch bei HA zu finden sein.
